src/index.ts

```diff
@@ -109,130 +131,211 @@ const FORWARD_RESPONSE_HEADERS = [
   'x-xss-protection',
 ];
 
+function restResolver(
```

Comment:
Can we move these to different files ? And I see there is some common code between the two can we do reuse via functions or class inheritance ?